File: src/creational/prototype/typescript/classes/interfaces/car-protocol.ts
import { IdsKeys } from '../car-cache';

type ID = number | null;
type TYPE = IdsKeys | null;

export abstract class Car {
  [x: string]: any;
  constructor(protected _type: TYPE = null, protected _id: ID = null) {
    let _ = 0;
    for (let i = 0; i < 100000; i++)
      for (let j = 0; j < 10000; j++)
        if (i != 0 && j != 0 && i % j != 0) _ += i * j;
    console.log(' > Creating new car:', this.__proto__.constructor.name);
  }

  set type(type: TYPE) {
    if (type === null) return;
    this._type = type;
  }

  get type(): TYPE {
    return this._type;
  }

  set id(id: ID) {
    if (typeof id !== 'number') return;
    this._id = id;
  }

  get id(): ID {
    return this._id;
  }

  abstract clone(): this;
}

File: src/creational/prototype/python/classes/interfaces/car_protocol.py
from abc import ABCMeta, abstractmethod
from copy import copy


class Car(metaclass=ABCMeta):

    def __init__(self):

        for i in range(10000):
            for n in range(1000):
                if i != 0 and n != 0 and i % n != 0:
                    _ = i*n

        print(' > Creating new car:', self.__class__.__name__)
        self.id = None
        self.type = None

    @abstractmethod
    def display_name(self) -> None:
        pass

    def set_type(self, type: str) -> None:
        self.type = type

    def get_type(self) -> str:
        return self.type

    def get_id(self) -> str:
        return self.id

    def set_id(self, id) -> None:
        self.id = id

    def clone(self) -> object:
        return copy(self)
